api/v1/helpers.go

```diff
@@ -567,7 +567,10 @@ func (v *VerticaDB) GetUpgradePolicyToUse() UpgradePolicyType {
 		const ceLicenseLimit = 3
 		if vinf.IsEqualOrNewer(OnlineUpgradeVersion) &&
 			!v.IsKSafety0() &&
-			(v.getNumberOfNodes() > ceLicenseLimit || v.Spec.LicenseSecret != "") {
+			(v.getNumberOfNodes() > ceLicenseLimit || v.Spec.LicenseSecret != "") &&
+			// online upgrade is not allowed if a there is already a sandbox
+			// in vertica
+			len(v.Status.Sandboxes) == 0 {
```

Comment:
I think `len(v.Status.Sandboxes) == 0`. sandboxing and upgrade cannot happen at the same time because the reconcile loop is sequential and upgrade comes before sandboxing. `v.Status.Sandboxes` is what tell us there is an actual sandbox. Moreover, if I remove `Spec.Sandboxes` from vdb it does not mean the sandbox is gone so the operator might think doing online upgrade is fine when it is not.

If I change the image and add a sandbox to `Spec.Sandboxes`, it is going to online upgrade first and sandbox after, which is fine.

Reply:
If sandbox reconciler fails before/when updating the status, we will run online upgrade which might not be good. Probably we can do len(v.Spec.Sandboxes) == 0 && len(v.Status.Sandboxes) == 0.